feature/experimental/bgp/otg_tests/bgp_always_compare_med/bgp_always_compare_med_test.go

```diff
@@ -0,0 +1,583 @@
+	otg.PushConfig(t, config)
+	time.Sleep(30 * time.Second)
+	otg.StartProtocols(t)
+	time.Sleep(30 * time.Second)
+
+	return config
+}
+
+// verifyTraffic confirms that every traffic flow has the expected amount of loss (0% or 100%
+// depending on wantLoss, +- 2%).
+func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, c gosnappi.Config, flowName string, wantLoss bool) {
+	otg := ate.OTG()
+	otgutils.LogFlowMetrics(t, otg, c)
+	t.Logf("Verifying flow metrics for flow %s\n", flowName)
+	recvMetric := gnmi.Get(t, otg, gnmi.OTG().Flow(flowName).State())
+	txPackets := recvMetric.GetCounters().GetOutPkts()
+	rxPackets := recvMetric.GetCounters().GetInPkts()
+	lostPackets := txPackets - rxPackets
+	lossPct := lostPackets * 100 / txPackets
+	if wantLoss {
+		if lossPct < 100-tolerancePct {
+			t.Errorf("Traffic is expected to fail %s\n got %v, want 100%% failure", flowName, lossPct)
+		} else {
+			t.Logf("Traffic Loss Test Passed!")
+		}
+	} else {
+		if lossPct > tolerancePct {
+			t.Errorf("Traffic Loss Pct for Flow: %s\n got %v, want 0", flowName, lossPct)
+		} else {
+			t.Logf("Traffic Test Passed!")
+		}
+	}
+}
+
+// setMED is used to configure routing policy to set BGP MED on DUT.
+func setMED(t *testing.T, dut *ondatra.DUTDevice, d *oc.Root) {
+	// Configure SetMED on DUT.
+	rp := d.GetOrCreateRoutingPolicy()
+
+	pdef1 := rp.GetOrCreatePolicyDefinition(setMEDPolicy100)
+	actions1 := pdef1.GetOrCreateStatement(aclStatement20).GetOrCreateActions()
+	actions1.GetOrCreateBgpActions().SetMed = oc.UnionUint32(bgpMED100)
+
+	pdef2 := rp.GetOrCreatePolicyDefinition(setMEDPolicy50)
+	actions2 := pdef2.GetOrCreateStatement(aclStatement20).GetOrCreateActions()
+	actions2.GetOrCreateBgpActions().SetMed = oc.UnionUint32(bgpMED50)
+
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+
+	// Apply setMed import policy on eBGP Peer1 - OTG Port2 - with MED 100.
+	dutPolicyConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).
+		Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp().
+		PeerGroup(peerGrpName2).ApplyPolicy().ImportPolicy()
+	gnmi.Replace(t, dut, dutPolicyConfPath.Config(), []string{setMEDPolicy100})
+
+	// Apply setMed Import policy on eBGP Peer2 - OTG Port3 -  with MED 50.
+	dutPolicyConfPath = gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).
+		Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp().
+		PeerGroup(peerGrpName3).ApplyPolicy().ImportPolicy()
+	gnmi.Replace(t, dut, dutPolicyConfPath.Config(), []string{setMEDPolicy50})
+}
+
+// verifySetMed is used to validate MED on received prefixes at OTG Port1.
+func verifySetMed(t *testing.T, otg *otg.OTG, config gosnappi.Config, wantMEDValue uint32) {
+	_, ok := gnmi.WatchAll(t,
+		otg,
+		gnmi.OTG().BgpPeer(ateSrc.Name+".BGP4.peer").UnicastIpv4PrefixAny().State(),
+		time.Minute,
+		func(v *ygnmi.Value[*otgtelemetry.BgpPeer_UnicastIpv4Prefix]) bool {
+			_, present := v.Val()
```

Comment:
you can just do "return val.IsPresent()"

Reply:
updated